Lib/zoneinfo/_tzpath.py

```diff
@@ -102,6 +102,71 @@ def _validate_tzfile_path(path, _base=_TEST_PATH):
 del _TEST_PATH
 
 
+def available_timezones():
+    """Returns a set containing all available time zones.
+
+    .. caution::
+
+        This may attempt to open a large number of files, since the best way to
+        determine if a given file on the time zone search path is to open it
+        and check for the "magic string" at the beginning.
+    """
+    from importlib import resources
+
+    valid_zones = set()
+
+    # Start with loading from the tzdata package if it exists: this has a
+    # pre-assembled list of zones that only requires opening one file.
+    try:
+        with resources.open_text("tzdata", "zones") as f:
+            for zone in f:
+                zone = zone.strip()
+                if zone:
+                    valid_zones.add(zone)
```

Comment:
I am not sure if it's better, but it could be replaced with

```python
valid_zones.update(filter(None, map(str.strip, f)))
```

Reply:
I personally like using these more functional constructs like maps and filters, but I don't think this will make it appreciably faster (probably a little, but I'm guessing it won't shave off more than a few µs) and I think most people would find it harder to read and understand the filter-map version.

Also, for debugging reasons, it's nice when the different actions are all separate lines.

I think if this were a smaller function I'd probably go with your version.